src/shared/src/types/address_type.rs

```diff
@@ -0,0 +1,61 @@
+use std::fmt;
+
+use candid::{CandidType, Deserialize};
+
+use super::account::TokenAccountId;
+
+#[derive(CandidType, Deserialize, Clone, Debug, Eq, PartialEq)]
```

Comment:
This refers specifically to account addresses, not canister addresses or other sorts of address?  If so, something like `AccountIdType` would be less ambiguous.  A rustdoc would also help clarify the purpose.

```suggestion
/// Token account ID types.
#[derive(CandidType, Deserialize, Clone, Debug, Eq, PartialEq)]
```